### PR TITLE
Allow CORS requests in Secret Store API

### PR DESCRIPTION
**Pull request by adetante**
On the same model as what is done for JSON-RPC, this PR allows to specify the list of domains to allow for cross-domain requests to Secret Store API.

With CLI flag ` --secretstore-http-cors=http://mydomain` or in the toml configuration file.

The default behavior is the same as the current one: all cross-domain requests are rejected.

### CODE REVIEW

File: secret-store/src/listener/http_listener.rs [hunk 1]
```diff
@@ -204,6 +211,24 @@ impl KeyServerHttpHandler {
 			},
 		}
 	}
+
+	fn cors(&self, req: &HttpRequest<Body>) -> Cors {
```
Comment (svyatonik) [inline secret-store/src/listener/http_listener.rs]: Could you, please, remove trailing tabs?

File: secret-store/src/listener/http_listener.rs [hunk 2]
```diff
@@ -204,6 +211,24 @@ impl KeyServerHttpHandler {
 			},
 		}
 	}
+
+	fn cors(&self, req: &HttpRequest<Body>) -> Cors {		
+		match req.headers().get(header::ORIGIN) {
+			None => Cors::CorsAuthorized(None),
+			Some(origin) => {
+				let is_authorized = self.handler.cors.clone()
```
Comment (svyatonik) [inline secret-store/src/listener/http_listener.rs]:
```suggestion
				let is_authorized = self.handler.cors.as_ref()
```

File: secret-store/src/listener/http_listener.rs [hunk 3]
```diff
@@ -213,69 +238,75 @@ impl Service for KeyServerHttpHandler {
...
+						warn!(target: "secretstore", "Ignoring invalid {}-request {}", req_method, req_uri);
+						HttpResponse::builder()
+							.status(HttpStatusCode::NOT_FOUND)
+							.body(Body::empty())
+							.expect("Nothing to parse, cannot fail; qed")
+					}
+				}))
 			}
-		}))
+		}
```
Comment (svyatonik) [inline secret-store/src/listener/http_listener.rs]: Trailing tabs

**Comment by adetante** [comment]
You're right: I'm going to change my implementation to the jsonrpc-server-utils crate.
Thank you for the feedback.

**Comment by adetante** [comment]
I updated based on your comments.
Thank you for reviewing

**Comment by svyatonik** [comment]
Test failure seems unrelated - SS tests are working fine on my laptop.

**Pull request merged**